src/IcedTasks/CancellableTaskBuilderBase.fs

```diff
@@ -48,6 +48,33 @@ module CancellableTaskBase =
     and CancellableTaskBaseCode<'TOverall, 'T, 'Builder> =
         ResumableCode<CancellableTaskBaseStateMachineData<'TOverall, 'Builder>, 'T>
 
+    let inline yieldOnBindLimit (code: CancellableTaskBaseCode<_, _, _>) =
+        CancellableTaskBaseCode(fun sm ->
+            let mutable __stack_code_continue = true
+
+            if Trampoline.Current.CheckBindLimit() then
+                let __stack_yield_fin = ResumableCode.Yield().Invoke(&sm)
+
+                if __stack_yield_fin then
+                    __stack_code_continue <- true
+                else
+                    let mutable __stack_awaiter = Trampoline.Current.Awaiter
+
+                    MethodBuilder.AwaitUnsafeOnCompleted(
+                        &sm.Data.MethodBuilder,
+                        &__stack_awaiter,
+                        &sm
+                    )
+
+                    __stack_code_continue <- false
+
+            if __stack_code_continue then
+                let __stack_code_fin = code.Invoke(&sm)
+                __stack_code_fin
+            else
+                false
+        )
```

Comment:
This can be simpler but the compiler is very finicky about resumable code.
The worst thing I've seen, definitely a bug, is compiling without warnings but then calling dynamic implementations at runtime.